### PR TITLE
[csharp-netcore] Adds ability to inherit api

### PR DESCRIPTION
**Pull request by devhl-labs**
Adds the ability to inherit the api, plus other refactors and bug fixes.

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java [hunk 1]
```diff
@@ -505,10 +505,6 @@ public boolean getRequired() {
         return required;
     }
 
-    public boolean compulsory() {
-        return getRequired() && !isNullable;
-    }
-
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java:511]: @devhl-labs thanks for the PR.

I think that's a breaking change.

shall we keep it for the time being and remove it in 7.0.0 release later?

cc @OpenAPITools/generator-core-team

Comment (devhl-labs) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java:511]: I added that somewhat recently and nothing is using it in master. Do you want it left still?

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java]: If it has not been included in any stable release, I'm happy to have it removed (or renamed)

Comment (devhl-labs) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java:511]: Okay great, I've renamed it.

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team as the change impacts default codegen.

**Comment by wing328** [comment]
All tests passed via https://ci.appveyor.com/project/WilliamCheng/openapi-generator/builds/45628055

Will merge it later today if no one has further feedback on this PR.

**Pull request merged**